src/Network/Wai/Middleware/Servant/Errors.hs

```diff
@@ -187,11 +188,10 @@ responseBody res =
 -- Its used in the library provided 'HasErrorBody' /JSON/ instance
 encodeAsJsonError :: ErrorLabels -> StatusCode -> ErrorMsg -> LB.ByteString
 encodeAsJsonError ErrorLabels {..} code content =
-  encode $ Object
-         $ H.fromList
-           [ (errName, String $ unErrorMsg content)
-           , (errStatusName, Number $ toScientific code )
-           ]
+  encode $ toJSON $ H.fromList
+    [ (errName, toJSON (unErrorMsg content))
```

Comment:
Suggestion: (Not merge blocking): This can be simplified further by using `.=`  from `Data.Aeson`
i.e 

```
[ errName .= unErrorMsg content
, errStatusName .= toScientific code
]

```

Reply:
I tried to do this before (i.e use `(.=)`), but it didn't compile for me w/ `aeson` 2.0. [The `(.=)` was changed to be of type `Key`](https://hackage.haskell.org/package/aeson-2.0.3.0/docs/Data-Aeson.html#v:.-61-), and the fields in `ErrorLabels` are not of type `Key` (making them `Key` would not be backwards compatible either). My initial attempt used CPP to get around this, but it felt hacky, so I just settled on bumping the lower bound because everybody has been doing that to get around the breaking changes. I saw later that liberal usage of `toJSON` can let us be backwards compatible.

Reply:
I hadn't thought about the incompatibility with `Key`. Thanks for explaining.